bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/api/Authorization.java

```diff
@@ -0,0 +1,533 @@
+        loginRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
+        loginRequest.header(HttpHeader.ACCEPT, CONTENT_TYPE_JSON + ", text/plain, */*");
+        loginRequest.header(HttpHeader.CONTENT_TYPE, CONTENT_TYPE_JSON);
+        loginRequest.header(HttpHeader.ORIGIN, baseUrl);
+        loginRequest.header(HttpHeader.REFERER, baseUrl + "/ciam/auth/login");
+
+        String rid = generateCodeVerifier(24);
+        JSONObject loginContent = new JSONObject();
+        loginContent.put("username", config.email);
+        loginContent.put("password", config.password);
+        loginContent.put("rememberMe", false);
+        loginContent.put("rid", rid);
+        loginRequest.content(new StringContentProvider(loginContent.toString(), "utf-8"));
+
+        String preLoginToken = null;
+        ContentResponse loginResponse = send(loginRequest);
+        int status = loginResponse.getStatus();
+        String loginResponseString = loginResponse.getContentAsString();
+        logger.trace("Step 4: Login {} - {}", status, loginResponseString);
+        if (status == 200) {
+            JSONObject loginResponseJSON = new JSONObject(loginResponseString);
+            preLoginToken = loginResponseJSON.optString("token", null);
+        }
+        if (preLoginToken != null) {
+            return preLoginToken;
+        } else {
+            throw new MercedesMeAuthException("Failed to login. HTTP " + status + " " + loginResponseString);
+        }
+    }
+
+    /**
+     * Perform login with token to receive authorization code.
+     *
+     * @throws MercedesMeAuthException if response status isn't correct or code not delivered
+     * @throws MercedesMeApiException if an error occurs during API call
+     */
+    private String resumeAuthentication(HttpClient loginHttpClient, String resumeUrl, String preLoginToken)
+            throws MercedesMeAuthException, MercedesMeApiException {
+        String code = null;
+        MultiMap<@Nullable String> authParams = new MultiMap<>();
+        authParams.add("token", preLoginToken);
+
+        Request authRequest = loginHttpClient.POST(baseUrl + resumeUrl).followRedirects(false);
+        authRequest.header(HttpHeader.USER_AGENT, Constants.AUTH_USER_AGENT);
+        authRequest.header(HttpHeader.ACCEPT_LANGUAGE, Constants.AUTH_LANGUAGE);
+        authRequest.header(HttpHeader.ACCEPT, "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8");
+        authRequest.header(HttpHeader.CONTENT_TYPE, CONTENT_TYPE_URL_ENCODED);
+        authRequest.header(HttpHeader.ORIGIN, baseUrl);
+        authRequest.header(HttpHeader.REFERER, baseUrl + "/ciam/auth/login");
+        authRequest.content(new StringContentProvider(CONTENT_TYPE_URL_ENCODED,
+                UrlEncoded.encode(authParams, StandardCharsets.UTF_8, false), StandardCharsets.UTF_8));
+
+        ContentResponse authResponse = send(authRequest);
+        int status = authResponse.getStatus();
+        if (status < 400) {
```

Comment:
```suggestion
        if (status < HttpStatus.BAD_REQUEST_400) {
```

Does it matter if Informational 1xx, Successful 2xx or Redirection 3xx is returned? Perhaps a more direct check would make sense, e.g.:

```suggestion
        if (!HttpStatus.isSuccess(status) {
```

or:

```suggestion
        if (!HttpStatus.isRedirection(status) {
```